CMakeLists.txt

```diff
@@ -1,60 +1,85 @@
 )
 
-include_directories(include)
+## Compile as C++11, supported in ROS Kinetic and newer
+add_definitions(-std=c++11 -Wall -Wextra -Wno-switch)
 
-list(REMOVE_ITEM SOURCES 
-        ${CMAKE_CURRENT_SOURCE_DIR}/src/TimeStamp.cpp
+## Find catkin macros and libraries
+## if COMPONENTS list like find_package(catkin REQUIRED COMPONENTS xyz)
+## is used, also find other catkin packages
+find_package(catkin REQUIRED COMPONENTS
+  sensor_msgs
+  roscpp
 )
+find_package (Threads REQUIRED)
 
-set(HEADER_FILES
-  "include/ahrs/AHRS.h"
-  "include/ahrs/AHRSProtocol.h"
-  "include/ahrs/ContinuousAngleTracker.h"
-  "include/ahrs/IBoardCapabilities.h"
-  "include/ahrs/IIOCompleteNotification.h"
-  "include/ahrs/IIOProvider.h"
-  "include/ahrs/IMUProtocol.h"
-  "include/ahrs/IMURegisters.h"
-  "include/ahrs/InertialDataIntegrator.h"
-  "include/ahrs/ITimestampedDataSubscriber.h"
-  "include/ahrs/OffsetTracker.h"
-  "include/ahrs/SerialIO.h"
-  "include/ahrs/SerialPort.h")
-
-add_library(ahrs SHARED ${SOURCES})
-set_target_properties(ahrs PROPERTIES 
-  PUBLIC_HEADER "${HEADER_FILES}"
-  FRAMEWORK TRUE
+###################################
+## catkin specific configuration ##
+###################################
+## The catkin_package macro generates cmake config files for your package
+## Declare things to be passed to dependent projects
+## INCLUDE_DIRS: uncomment this if your package contains header files
+## LIBRARIES: libraries you create in this project that dependent projects also need
+## CATKIN_DEPENDS: catkin_packages dependent projects also need
+## DEPENDS: system dependencies of this project that dependent projects also need
+catkin_package(
+  INCLUDE_DIRS include
+  LIBRARIES navx
+  CATKIN_DEPENDS sensor_msgs
+  CATKIN_DEPENDS roscpp
+#  DEPENDS system_lib
+  DEPENDS Threads
+)
+
+###########
+## Build ##
+###########
+
+file(GLOB SOURCES 
+        "${CMAKE_CURRENT_SOURCE_DIR}/src/*.cpp"
+        #"${CMAKE_CURRENT_SOURCE_DIR}/include/ahrs/*.h"
 )
 
-add_executable(NavXTimeStamp
-        "${CMAKE_CURRENT_SOURCE_DIR}/src/TimeStamp.cpp"
+## Specify additional locations of header files
+## Your package locations should be listed before other locations
+include_directories(
+ include
+  ${catkin_INCLUDE_DIRS}
 )
-target_link_libraries(NavXTimeStamp ahrs ${CMAKE_THREAD_LIBS_INIT})
-#target_compile_features(NavXTimeStamp PRIVATE cxx_range_for)
 
-install(TARGETS ahrs 
-        LIBRARY DESTINATION /usr/local/lib
-        PUBLIC_HEADER DESTINATION /usr/local/include/ahrs
+## Declare a C++ library
+add_library(navx
+ ${SOURCES}
 )
+add_dependencies(navx ${${PROJECT_NAME}_EXPORTED_TARGETS} ${catkin_EXPORTED_TARGETS})
+
+add_executable(${PROJECT_NAME}_node src/navx_node.cpp)
+
+## Add cmake target dependencies of the executable
+## same as for the library above
+add_dependencies(navx_node ${${PROJECT_NAME}_EXPORTED_TARGETS} ${catkin_EXPORTED_TARGETS})
+
+## Specify libraries to link a library or executable target against
```

Comment:
Delete the lines between add_executable/add_deps/target_link_libraries, and make them each one line. Wherever it says ${PROJECT_NAME} make it hardcoded navx.